### PR TITLE
🔨 Add key binding related functions in API

### PR DESCRIPTION
**Pull request by Florian-1801**

### CODE REVIEW

File: src/api/players/client/canUseRegisteredKey.lua [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+            if (ButtonIndex) then
+                if (v.key == string.upper(ButtonIndex)) then
+                    v.canUse = Bool
+                end
+            else
+                v.canUse = Bool
+            end
+        end
+    end
+end
```
Comment (PABLO-1610) [inline src/api/players/client/canUseRegisteredKey.lua:16]: Please consider using camelCase for variable names.

image

File: src/api/players/client/keyList.lua [hunk 2]
```diff
@@ -0,0 +1,8 @@
+KeyList = { ["ESC"] = 322, ["F1"] = 288, ["F2"] = 289, ["F3"] = 170, ["F5"] = 166, ["F6"] = 167, ["F7"] = 168, ["F8"] = 169, ["F9"] = 56,
+            ["F10"] = 57, ["~"] = 243, ["1"] = 157, ["2"] = 158, ["3"] = 160, ["4"] = 164, ["5"] = 165, ["6"] = 159, ["7"] = 161, ["8"] = 162,
+            ["9"] = 163, ["-"] = 84, ["="] = 83, ["BACKSPACE"] = 177, ["TAB"] = 37, ["Q"] = 44, ["W"] = 32, ["E"] = 38, ["R"] = 45, ["T"] = 245,
+            ["Y"] = 246, ["U"] = 303, ["P"] = 199, ["["] = 39, ["]"] = 40, ["ENTER"] = 18, ["CAPS"] = 137, ["A"] = 34, ["S"] = 8, ["D"] = 9,
+            ["F"] = 23, ["G"] = 47, ["H"] = 74, ["K"] = 311, ["L"] = 182, ["LEFTSHIFT"] = 21, ["Z"] = 20, ["X"] = 73, ["C"] = 26, ["V"] = 0,
+            ["B"] = 29, ["N"] = 249, ["M"] = 244, [","] = 82, ["."] = 81, ["LEFTCTRL"] = 36, ["LEFTALT"] = 19, ["SPACE"] = 22, ["RIGHTCTRL"] = 70,
+            ["HOME"] = 213, ["PAGEUP"] = 10, ["PAGEDOWN"] = 11, ["DELETE"] = 178, ["LEFT"] = 174, ["RIGHT"] = 175, ["TOP"] = 27, ["DOWN"] = 173,
+            ["NENTER"] = 201, ["N4"] = 108, ["N5"] = 60, ["N6"] = 107, ["N+"] = 96, ["N-"] = 97, ["N7"] = 117, ["N8"] = 61, ["N9"] = 118, }
```
Comment (PABLO-1610) [inline src/api/players/client/keyList.lua:8]: Do not implements this list, we'll use number instead.

File: src/api/players/client/onControlPress.lua [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                _NCS:trace("no action defined", 4)
+            end
+            return true
+        end
+    else
+        _NCS:trace("key not defind", 4)
+    end
+end
+
+
```
Comment (PABLO-1610) [inline src/api/players/client/onControlPress.lua:20]: Impossible due to previous comment

**Review by PABLO-1610 — CHANGES_REQUESTED** [review]

**Comment by Florian-1801** [comment]
I think is good now

**Comment by PABLO-1610** [comment]
image

**Comment by Florian-1801** [comment]
mybad, it's fix

**Review by PABLO-1610 — CHANGES_REQUESTED** [review]
Instead of doing all this code, you should probably look at [RegisterKeyMapping](https://docs.fivem.net/natives/?_0xD7664FD1).

**Comment by Florian-1801** [comment]
Yes, I think I will rework this code later, if you want to delete it, let's do it

**Comment by PABLO-1610** [comment]
> Yes, I think I will rework this code later, if you want to delete it, let's do it

I don't need to delete it, it's your work but i think that key mappings are better.

**Comment by Florian-1801** [comment]
I just use key mapping but I implement a system to disable or enables it

**Comment by Florian-1801** [comment]
But if you think is better, I will do what you want

**Comment by Florian-1801** [comment]
you want to close the pull request or push it ?

**Pull request closed without merging**